### DISCUSSION TITLE
String is truncated at colon

### DISCUSSION THREAD

**Discussion by johanskoldekrans** [ravendb/ravendb#15272] [Q&A]
I came across a strange error that I haven’t found an explanation for. I have saved an entity in the database that has a string with a colon in it. When I read from it from the database I get everything up until the colon, the rest is missing. 

Should I escape the colon or set an attribute on the property?!

View of document in database:
![image](https://user-images.githubusercontent.com/18699147/201863668-0ecd35b0-5cdd-401a-9ad3-f4c3ba450e00.png)

Code fetching the document:
![image](https://user-images.githubusercontent.com/18699147/201863728-ae924412-2d8d-4b9a-9bef-15fbff6a81de.png)

Running test and debugging:
![image](https://user-images.githubusercontent.com/18699147/201863783-daa93bb2-e25e-41da-8266-ea713bbcc331.png)

**Comment by johanskoldekrans** [comment]
The class used is just a plain class with getters and setters to avoid other problems that might interfere.

**Comment by ayende** [comment]
This should absolutely not happen. You shouldn't escape the colon.
Can you do a small repro for this?

**Reply by mcrio** [reply]
..and in meantime maybe also paste here the `Country` class, seems like there is more going on in this class and also `Alpha2` is nulled after loading the doc.

**Reply by johanskoldekrans** [reply]
BaseClass only has a string Id

    ```
public class Country : Entity
    {
        public Country
        (
            string id,
            string name,
            string alpha3,
            string number
        ) : base(id)
        {
            Name = name;
            Alpha2 = id;
            Alpha3 = alpha3;
            Number = number;
            Iso3166 = $"ISO 3166-2:{id}";
        }
        public string Name { get; }
        public string Alpha2 { get; }
        public string Alpha3 { get; }
        public string Number { get; }
        public string Iso3166 { get; }

        public bool IsSanctioned { get; set; }

        [JsonIgnore]
        public RiskLevelEnum RiskLevel => RiskLevelEnum.FromValue(RiskLevelId);

        public ushort RiskLevelId { get; set; } = RiskLevelEnum.NotSet.Value;

        [JsonIgnore]
        public RiskIndicatorEnum RiskIndicator => RiskIndicatorEnum.FromValue(RiskIndicatorId);

        public ushort RiskIndicatorId { get; set; } = RiskIndicatorEnum.NotSet.Value;
    }
```

**Reply by ayende** [reply]
You are missing a parameter less ctor. That is likely the problem.
The serializer is calling to the `Country(...)` ctor and passing empty / null `id` (since that isn't set as the id.

**Reply by johanskoldekrans** [reply]
Yes, that seems to have been the problem. New class looks like:
public class Country : Entity
    {
        public Country()
        {

        }
        public Country
        (
            string id,
            string name,
            string alpha3,
            string number
        ) : base(id)
        {
            Name = name;
            Alpha3 = alpha3;
            Number = number;
            Iso3166 = $"ISO 3166-2:{id}";
        }
        public string Name { get; init; }
        public string Alpha2 => Id;
        public string Alpha3 { get; init; }
        public string Number { get; init; }
        public string Iso3166 { get; init; }

        public bool IsSanctioned { get; set; }

        [JsonIgnore]
        public RiskLevelEnum RiskLevel => RiskLevelEnum.FromValue(RiskLevelId);

        public ushort RiskLevelId { get; set; } = RiskLevelEnum.NotSet.Value;

        [JsonIgnore]
        public RiskIndicatorEnum RiskIndicator => RiskIndicatorEnum.FromValue(RiskIndicatorId);

        public ushort RiskIndicatorId { get; set; } = RiskIndicatorEnum.NotSet.Value;
    }

**Comment by johanskoldekrans** [comment]
[TestProject.zip](https://github.com/ravendb/ravendb/files/10013322/TestProject.zip)

Uploading my testproject if anyone wants to look.

**Reply by johanskoldekrans** [reply]
I could actually remove the empty constructor after fixing init on the properties.

**Reply by ayende** [reply]
Yes, if you have the `init` there, that would make the serializer manage everything properly
